svm/src/transaction_processor.rs

```diff
@@ -208,11 +208,12 @@ impl<FG: ForkGraph> TransactionBatchProcessor<FG> {
     }
 
     /// Returns the current environments depending on the given epoch
-    pub fn get_environments_for_epoch(&self, epoch: Epoch) -> ProgramRuntimeEnvironments {
+    /// Returns None if the call could result in a deadlock
+    pub fn get_environments_for_epoch(&self, epoch: Epoch) -> Option<ProgramRuntimeEnvironments> {
```

Comment:
Looks like you got things working with `Bank::load_program()` for `dev-context-utils-only`, I think we can do it for this function too. Something like below to limit the pub-ness of this function unless we're using `dev-context-only-utils`:
https://github.com/anza-xyz/agave/blob/e804bcce291e517442515f12e2b47298e8b6f4c1/runtime/src/installed_scheduler_pool.rs#L283-L284

Reply:
Yes, this seems to work. Pushed an update to the PR.